src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -1662,8 +1688,14 @@ int D4NFilterObject::D4NFilterDeleteOp::delete_obj(const DoutPrefixProvider* dpp
 	block.size = static_cast<uint64_t>(cur_len);
 
 	if ((ret = blockDir->get(dpp, &block, y)) < 0) {
-	  ldpp_dout(dpp, 0) << "Failed to retrieve directory entry for: " << source->get_name() << " blockid: " << fst << " block size: " << cur_len << ", ret=" << ret << dendl;
-	  return ret;
+          if (ret == -ENOENT) {
+	    ldpp_dout(dpp, 0) << "Directory entry for: " << source->get_name() << " blockid: " << fst << " block size: " << cur_len << " does not exist; continuing" << dendl;
+            fst += g_conf()->rgw_get_obj_max_req_size; // does this handle range requests? -Sam
```

Comment:
I don't think this is correct, I had explained in my comment in your previous PR how to correctly increment fst

Reply:
May you please point me to that information? I attempted to search for it but wasn't able to find it

Reply:
Check this: https://github.com/samarahu/ceph/commit/d75a3903e04cabf8059477bdee2fa7fea7f7961c#diff-5cf52d0baced08513b69bbb3cf4a29682289fce58451b9567c6d2e86ff080bceR1668